### PR TITLE
[flink] Support minor compact strategy for dedicated compaction action.

### PR DESCRIPTION
**Pull request by LinMingQiang**
### Purpose

Linked issue: (https://github.com/apache/paimon/issues/4566)

### Tests
```

- `--compact_strategy` Determines how to pick files to be merged, the default is determined by the runtime execution mode, streaming-mode use minor strategy and batch-mode use full strategy.

- `full `: Only supports batch mode. All files will be selected for merging.

- `minor `: Pick the set of files that need to be merged based on specified conditions.

```

### CODE REVIEW

File: docs/content/maintenance/dedicated-compaction.md [hunk 1]
```diff
@@ -107,6 +107,7 @@ Run the following command to submit a compaction job for the table.
     --database <database-name> \ 
     --table <table-name> \
     [--partition <partition-name>] \
+    [--compact_strategy <minor / full>] \
```
Comment (wwj6591812) [inline docs/content/maintenance/dedicated-compaction.md:110]: Now in paimon, there are already three compact_strategy, FullCompaction, UniversalCompaction, ForceUpLevel0Compaction.
I think here we change compact_strategy to compact_type may better.?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 2]
```diff
@@ -59,6 +59,8 @@ public class CompactAction extends TableActionBase {
 
     @Nullable private Duration partitionIdleTime = null;
 
+    private Boolean fullCompaction;
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:62]: Why not define a enum of CompactionType?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactActionFactory.java [hunk 3]
```diff
@@ -76,6 +76,10 @@ public Optional<Action> create(MultipleParameterToolAdapter params) {
                 action.withPartitionIdleTime(
                         TimeUtils.parseDuration(params.get(PARTITION_IDLE_TIME)));
             }
+            String compactStrategy = params.get(COMPACT_STRATEGY);
+            if (compactStrategy != null) {
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactActionFactory.java:80]: compact_type?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTablesStoreCompactOperator.java [hunk 4]
```diff
@@ -165,7 +168,8 @@ public void processElement(StreamRecord<RowData> element) throws Exception {
                     files.isEmpty(),
                     "Batch compact job does not concern what files are compacted. "
                             + "They only need to know what buckets are compacted.");
-            write.compact(partition, bucket, true);
+            // `minor` compact strategy is supported in batch mode.
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTablesStoreCompactOperator.java:171]: `minor` and `full` compact type are both supported in batch mode.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CompactActionITCaseBase.java [hunk 5]
```diff
@@ -37,6 +42,12 @@
 /** Base IT cases for {@link CompactAction} and {@link CompactDatabaseAction} . */
 public class CompactActionITCaseBase extends ActionITCaseBase {
 
+    protected static final DataType[] FIELD_TYPES =
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CompactActionITCaseBase.java:45]: private

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CompactActionITCaseBase.java [hunk 6]
```diff
@@ -87,4 +98,29 @@ protected void checkFileAndRowSize(
         assertThat(files.size()).isEqualTo(fileNum);
         assertThat(count).isEqualTo(rowCount);
     }
+
+    protected void checkLatestSnapshot(
+            FileStoreTable table, long snapshotId, Snapshot.CommitKind commitKind) {
+        SnapshotManager snapshotManager = table.snapshotManager();
+        Snapshot snapshot = snapshotManager.snapshot(snapshotManager.latestSnapshotId());
```
Comment (wwj6591812) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CompactActionITCaseBase.java:105]: Unboxing of 'snapshotManager.latestSnapshotId()' may produce 'NullPointerException'

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTablesStoreCompactOperator.java (hunk 4 above)
Comment (LinMingQiang) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/MultiTablesStoreCompactOperator.java:171]: Yeah, this reminder should be placed above.  `full compact is not supported in streaming mode.`

File: docs/content/maintenance/dedicated-compaction.md (hunk 1 above)
Comment (LinMingQiang) [reply docs/content/maintenance/dedicated-compaction.md:110]: Essentially there are only two strategies `full` or `minor`, so using `compact_strategy ` is more suitable for this scenario.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java (hunk 2 above)
Comment (LinMingQiang) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:62]: Refer to above, If there are more other strategies in the future, we can use enumeration.

**Review by JingsongLi — COMMENTED** [review]
Can you also modify Flink & Spark procedures?

**Comment by LinMingQiang** [comment]
> Can you also modify Flink & Spark procedures?

I plan to complete this in another PR.

Plan to add a new procedure `minor_compact`, and provide some new parameters like 
`num-sorted-run.compaction-trigger` / `compaction.size-ratio`.

File: paimon-flink/paimon-flink-1.18/src/main/java/org/apache/paimon/flink/procedure/CompactProcedure.java [hunk 7]
```diff
@@ -125,6 +129,7 @@ public String[] call(
             ProcedureContext procedureContext,
             String tableId,
             String partitions,
+            String compactStrategy,
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-1.18/src/main/java/org/apache/paimon/flink/procedure/CompactProcedure.java:132]: It is better to keep compability?

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**